nni/trial.py

```diff
@@ -154,3 +156,22 @@ def report_final_result(metric):
         'value': to_json(metric)
     })
     platform.send_metric(metric)
+
+
+def _check_metric(metric):
+    """
+    Check if the metric is valid.
+
+    Parameters
+    ----------
+    metric:
+        serializable object.
+    """
+    if not isinstance(metric, (dict, int, float)):
+        raise ValueError('Metric should be a number or a dict with key "default" but got {type(metric)}')
+    if isinstance(metric, dict):
+        if 'default' not in metric.keys():
+            raise ValueError('Metric should have a key "default"')
+        for k, v in metric.items():
+            if not isinstance(v, (int, float)):
+                raise ValueError(f'Metric values should be numbers but got {type(v)} in key {k}')
```

Comment:
Can you add some unittests for this function?

It looks quite dangerous. For example, I'm not sure whether it works for types like `numpy.int32`, which I believe should be a valid case.